src/app/profile/settings/services/resource.service.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  ErrorHandler,
+  Inject,
+  Injectable,
+  OnDestroy
+} from '@angular/core';
+
+import { User, UserService } from 'ngx-login-client';
+import {
+  Observable,
+  ReplaySubject,
+  Subject,
+  Subscription
+} from 'rxjs';
+
+import { ScaledMemoryStat } from '../../../space/create/deployments/models/scaled-memory-stat';
+import { DeploymentApiService, EnvironmentStat } from '../../../space/create/deployments/services/deployment-api.service';
+
+const FAKE_SPACE_ID = 'f461b06c-3833-11e8-b467-0ed5f89f718b';
+
+enum CLASSES {
+  ICON_OK = 'pficon-ok',
+  ICON_WARN = 'pficon-warning-triangle-o',
+  ICON_ERR = 'pficon-error-circle-o'
+}
+
+export declare interface UsageSeverityEnvironmentStat {
+  environmentStat: EnvironmentStat;
+  iconClass: string;
+}
+
+@Injectable()
+export class ResourceService {
+
+  constructor(
+    private apiService: DeploymentApiService,
+    private userService: UserService
+  ) {}
+
+  public getEnvironmentsWithScaleAndIcon(): Observable<UsageSeverityEnvironmentStat[]> {
```

Comment:
Please open an issue to refactor this service class, or update one if the issue already exists. I'm surprised there's no test that calls `getEnvironmentsWithScaleAndIcon()`

If the other functions are indeed public, to be used elsewhere I'd like them to explicitly be public. If they're only intended as helpers for this public function, they should be made private.